src/ocm/components/AddHosts/AddHosts.tsx

```diff
@@ -44,8 +45,19 @@ const AddHosts: React.FC = () => {
   const handleHostsInstall = async () => {
     setSubmitting(true);
     try {
-      await HostsService.installAll(cluster);
-      void resetCluster();
+      if (!cluster.hosts) {
```

Comment:
For Day2, if you want to add more hosts, is not working in this way. (AddHosts class)
Maybe something like :
const hostsToBeInstalled = cluster.hosts.filter((host) =>
        canInstallHost(cluster, host.status),
      );
      if (hostsToBeInstalled && hostsToBeInstalled.length > 0) {
        await HostsService.installAll(hostsToBeInstalled);
        void resetCluster();
      } else {
        throw new Error(`Not all hosts are ready to be installed`);
      }

Reply:
Yes I agree, this part fall in between the gaps while cherry picking the commits.
I'm fixing it now. Just bear in mind the error message you put there is not quite accurate :)